lib/Connector/AlphaVantageConnector.php

```diff
@@ -0,0 +1,532 @@
+<?php
+/*
+ * Copyright (C) 2023 Xibo Signage Ltd
+ *
+ * Xibo - Digital Signage - http://www.xibo.org.uk
+ *
+ * This file is part of Xibo.
+ *
+ * Xibo is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * any later version.
+ *
+ * Xibo is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Xibo.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace Xibo\Connector;
+
+use Carbon\Carbon;
+use GuzzleHttp\Exception\GuzzleException;
+use Stash\Interfaces\PoolInterface;
+use Stash\Invalidation;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Xibo\Event\WidgetDataRequestEvent;
+use Xibo\Support\Exception\ConfigurationException;
+use Xibo\Support\Exception\GeneralException;
+use Xibo\Support\Exception\InvalidArgumentException;
+use Xibo\Support\Sanitizer\SanitizerInterface;
+use Xibo\Widget\Provider\DataProviderInterface;
+
+/**
+ * A connector to get data from the AlphaVantage API for use by the Currencies and Stocks Widgets
+ */
+class AlphaVantageConnector implements ConnectorInterface
+{
+    use ConnectorTrait;
+
+    public function registerWithDispatcher(EventDispatcherInterface $dispatcher): ConnectorInterface
```

Comment:
You can use `public function setFactories(ContainerInterface $container): ConnectorInterface` here to get the Cache Pool